web/src/popover_menus.js

```diff
@@ -633,6 +635,64 @@ export function initialize() {
         },
     });
 
+    register_popover_menu(".change_visibility_policy", {
+        placement: "bottom",
+        popperOptions: {
+            modifiers: [
+                {
+                    // The placement is set to bottom, but if that placement does not fit,
+                    // the opposite top placement will be used.
+                    name: "flip",
+                    options: {
+                        fallbackPlacements: ["top", "left"],
+                    },
+                },
+            ],
+        },
+        onShow(instance) {
+            popover_instances.change_visibility_policy = instance;
+            on_show_prep(instance);
+            const elt = $(instance.reference).closest(".change_visibility_policy").expectOne()[0];
+            const stream_id = $(elt).attr("data-stream-id");
+            const topic_name = $(elt).attr("data-topic-name");
+
+            instance.context =
+                popover_menus_data.get_change_visibility_policy_popover_content_context({
+                    stream_id,
+                    topic_name,
+                });
+            instance.setContent(
+                parse_html(render_change_visibility_policy_popover(instance.context)),
+            );
+        },
+        onMount(instance) {
+            const $popper = $(instance.popper);
+            const {stream_id, topic_name} = instance.context;
+
+            if (!stream_id) {
+                instance.hide();
+                return;
+            }
+
+            $popper.on("click", ".visibility_policy_option", (e) => {
+                $(".visibility_policy_option").removeClass("selected_visibility_policy");
+                $(e.currentTarget).addClass("selected_visibility_policy");
+
+                const visibility_policy = $(e.currentTarget).attr("data-visibility-policy");
+                user_topics.set_user_topic_visibility_policy(
+                    stream_id,
+                    topic_name,
+                    visibility_policy,
+                );
+                instance.hide();
```

Comment:
Just noting that this doesn't have a good way to do error handling -- it's probably OK as I don't think it will come up often enough to matter, but let's add a `// TODO: Figure out a good way to offer feedback if this request fails`.

Reply:
Comment added